nifi-nar-bundles/nifi-iaa-providers-bundle/nifi-file-identity-provider/src/main/java/org/apache/nifi/authentication/file/FileIdentityProvider.java

```diff
@@ -0,0 +1,216 @@
+    private PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+    private String identifier;
+
+    private static JAXBContext initializeJaxbContext() {
+        try {
+            return JAXBContext.newInstance(JAXB_GENERATED_PATH,  FileIdentityProvider.class.getClassLoader());
+        } catch (JAXBException e) {
+            throw new RuntimeException("Failed creating JAXBContext for " + FileIdentityProvider.class.getCanonicalName());
+        }
+    }
+
+    private static ValidationEventHandler defaultValidationEventHandler = new ValidationEventHandler() {
+        @Override
+        public boolean handleEvent(ValidationEvent event) {
+            return false;
+        }
+    };
+
+    static UserCredentialsList loadCredentialsList(String filePath) throws Exception {
+        return loadCredentialsList(filePath, defaultValidationEventHandler);
+    }
+
+    static UserCredentialsList loadCredentialsList(String filePath, ValidationEventHandler validationEventHandler) throws Exception {
+        final File userDetailsFile = new File(filePath);
+
+        if (userDetailsFile.exists()) {
+            final SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+            final Schema schema = schemaFactory.newSchema(UserCredentialsList.class.getResource(CREDENTIALS_XSD));
+
+            final Unmarshaller unmarshaller = JAXB_CONTEXT.createUnmarshaller();
+            unmarshaller.setSchema(schema);
+            unmarshaller.setEventHandler(validationEventHandler);
+            final JAXBElement<UserCredentialsList> element = unmarshaller.unmarshal(new StreamSource(userDetailsFile),
+                    UserCredentialsList.class);
+            UserCredentialsList credentialsList = element.getValue();
+            return credentialsList;
+        } else {
+            final String notFoundMessage = "The credentials configuration file was not found at: " +
+                    userDetailsFile.getAbsolutePath();
+            throw new FileNotFoundException(notFoundMessage);
+        }
+    }
+
+
+    @Override
+    public final void initialize(final LoginIdentityProviderInitializationContext initializationContext) throws ProviderCreationException {
+        this.identifier = initializationContext.getIdentifier();
+        this.issuer = getClass().getSimpleName();
+    }
+
+    @Override
+    public final void onConfigured(final LoginIdentityProviderConfigurationContext configurationContext) throws ProviderCreationException {
+        final Map<String, String> configProperties = configurationContext.getProperties();
+        for (String propertyKey : configProperties.keySet()) {
+            String propValue = configProperties.get(propertyKey);
+            logger.debug("found property '{}': '{}'", propertyKey, propValue);
+        }
+
+        credentialsFilePath = configProperties.get(PROPERTY_CREDENTIALS_FILE);
+        if (credentialsFilePath == null) {
```

Comment:
Can this return an empty String? `StringUtils.isEmpty()` may be a better check here than just `== null`.